plugins/inx/server_milestones.go

```diff
@@ -13,46 +13,26 @@ import (
 	"github.com/gohornet/hornet/pkg/model/storage"
 	"github.com/gohornet/hornet/pkg/tangle"
 	"github.com/iotaledger/hive.go/events"
-	"github.com/iotaledger/hive.go/serializer/v2"
 	"github.com/iotaledger/hive.go/workerpool"
 	inx "github.com/iotaledger/inx/go"
 )
 
 func milestoneForCachedMilestone(ms *storage.CachedMilestone) (*inx.Milestone, error) {
 	defer ms.Release(true) // milestone -1
 
-	milestone := ms.Milestone()
-
-	milestoneMsg := deps.Storage.CachedMessageOrNil(milestone.MessageID) // message + 1
-	if milestoneMsg == nil {
-		return nil, status.Errorf(codes.NotFound, "milestone message for %d not found", milestone.Index)
-	}
-	defer milestoneMsg.Release(true) // message -1
-
-	milestonePayload := milestoneMsg.Message().Milestone()
-	if milestone == nil {
-		return nil, status.Errorf(codes.Internal, "milestone message for %d does not contain a milestone", milestone.Index)
-	}
-	milestoneID, err := milestonePayload.ID()
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "error computing milestone ID: %s", err)
-	}
-
-	bytes, err := milestonePayload.Serialize(serializer.DeSeriModeNoValidation, nil)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "error serializing milestone payload: %s", err)
-	}
-
 	return &inx.Milestone{
-		MilestoneInfo: inx.NewMilestoneInfo(*milestoneID, uint32(milestone.Index), uint32(milestone.Timestamp.Unix())),
+		MilestoneInfo: inx.NewMilestoneInfo(
+			ms.Milestone().MilestoneID(),
+			uint32(ms.Milestone().Index()),
+			ms.Milestone().TimestampUnix()),
 		Milestone: &inx.RawMilestone{
-			Data: bytes,
+			Data: ms.Milestone().Data(),
 		},
 	}, nil
 }
 
 func milestoneForIndex(msIndex milestone.Index) (*inx.Milestone, error) {
-	cachedMilestone := deps.Storage.CachedMilestoneOrNil(msIndex) // milestone +1
+	cachedMilestone := deps.Storage.CachedMilestoneByIndexOrNil(msIndex) // milestone +1
```

Comment:
Do you want to add in `ReadMilestone(.0.)` below a check, that if `GetMilestoneIndex() == 0`, that you look it up by `GetMilestoneId()` in the `MilestoneRequest`?